db-schema.ts

```diff
@@ -0,0 +1,18 @@
+import {Schema} from 'mongoose';
+import mongoose = require('mongoose');
+
+
+const productSchema = new Schema({
+    name: String,
+    description: String,
+    price: Number,
+    quantity: Number,
+    imageUrl: String
+});
+
+const Product = mongoose.model('Product', productSchema);
```

Comment:
mongoose.model is a generic type - add type so you won't need so much `// @ts-ignore`s

Reply:
done